### PR TITLE
Add a AddonBrowserMapping model and admin

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-server/issues/20708

---

This PR adds both a model and the django admin for it. As mentioned in
the issue linked above, there is no specific permission for this admin
at the moment so admins should be the only ones allowed to access it.

### CODE REVIEW

File: src/olympia/addons/tests/test_admin.py [hunk 1]
```diff
@@ -1118,3 +1124,101 @@ def test_can_delete(self):
             f"[{restriction.addon.id}] deleted: ['FR']"
         )
         assert mail.outbox[0].to == ['amo-admins@mozilla.com']
+
+
+class TestAddonBrowserMappingAdmin(TestCase):
+    def setUp(self):
+        user = user_factory(email='someone@mozilla.com')
+        self.grant_permission(user, '*:*')
```
Comment (diox) [inline src/olympia/addons/tests/test_admin.py:1132]: We should start with a more specific permission. I suggest re-using `ADMIN_CURATION` (need to edit `DJANGO_PERMISSIONS_MAPPING`).

File: src/olympia/addons/admin.py [hunk 2]
```diff
@@ -649,6 +649,22 @@ def save_model(self, request, obj, form, change):
...
 
+@admin.register(models.AddonBrowserMapping)
+class AddonBrowserMappingAdmin(AMOModelAdmin):
+    list_display = ('addon__name', 'browser', 'extension_id', 'created', 'modified')
+    fields = ('addon', 'browser', 'extension_id')
+    raw_id_fields = ('addon',)
+    readonly_fields = ('created', 'modified')
+
+    def get_readonly_fields(self, request, obj=None):
+        return self.readonly_fields + (('addon',) if obj else ())
```
Comment (diox) [inline src/olympia/addons/admin.py:660]: Do we really care ? I understand this is not a change to be done lightly but ff there is a mistake it'd be nice to be able to correct it without having to delete and re-create the entry.

File: src/olympia/addons/models.py [hunk 3]
```diff
@@ -2385,3 +2386,24 @@ class Meta:
     def save(self, *args, **kwargs):
         self.hashed_guid = hashlib.sha256(self.guid.encode()).hexdigest()
         super().save(*args, **kwargs)
+
+
+class AddonBrowserMapping(ModelBase):
+    """
+    Mapping between a Firefox add-on on AMO and an extension ID for a different
+    browser.
```
Comment (diox) [inline src/olympia/addons/models.py:2394]: Nit: It's actually the other way around, right ? A mapping between an extension ID for a different browser to a Firefox Add-on on AMO

File: src/olympia/addons/models.py [hunk 4]
```diff
@@ -2385,3 +2386,24 @@ class Meta:
...
+        help_text='Add-on id this item will point to. If you do not know the '
+        'id, paste the slug instead and it will be transformed automatically '
+        'for you. You can also use the magnifying glass to see all the '
+        'available add-ons if you have access to the add-on admin page.',
+    )
+    browser = models.PositiveSmallIntegerField(choices=BROWSERS.items())
+    extension_id = models.CharField(max_length=255, null=False, db_index=True)
+
+    class Meta:
+        unique_together = ('addon', 'browser', 'extension_id')
```
Comment (diox) [inline src/olympia/addons/models.py:2409]: I think the actual unique constraint we want is `browser`, `extension_id`. We shouldn't need to include the `addon` in there, we never want the same browser + extension id combo to point to multiple add-ons.

(See also nitpicking above for the docstring)

**Comment by willdurand** [comment]
It's green again, ready for another review, thanks!

**Review by diox — APPROVED** [review]

**Pull request merged**